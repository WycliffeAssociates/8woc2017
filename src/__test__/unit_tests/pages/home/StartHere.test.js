import { shallow, mount, render } from "enzyme";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import StartHere from "../../../../js/pages/home/StartHere";
import toJson from 'enzyme-to-json';

describe("Home Component, StartHere", () => {
	const wrapper = shallow(<StartHere />);
	const tree = toJson(wrapper);
	
	it("is rendered", () => {
		expect(wrapper).toHaveLength(1);
	});
	it("has class, start-here", () => {
		expect(wrapper.hasClass("start-here")).toBe(true);
	});
	it("navigates to project page", () => {
		expect(wrapper.find("Link").props().to).toEqual("/projects");
	});
	it('has no changes', () => {
		expect(tree).toMatchSnapshot();
	});
});
