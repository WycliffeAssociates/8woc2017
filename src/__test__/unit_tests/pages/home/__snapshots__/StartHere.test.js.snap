// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component, StartHere has no changes 1`] = `
<div
  className="start-here"
>
  <Link
    replace={false}
    to="/projects"
  >
    <Button
      as="button"
      icon="right arrow"
      labelPosition="right"
    />
  </Link>
</div>
`;
